src/Helpers/ValueGetter.php

```diff
@@ -0,0 +1,60 @@
+<?php
+
+declare(strict_types=1);
+
+namespace SimPod\PhpSnmp\Helpers;
+
+use SimPod\PhpSnmp\Exception\NoSuchInstanceExists;
+use SimPod\PhpSnmp\Transport\SnmpClient;
+use function array_shift;
+use function array_values;
+use function assert;
+use function strpos;
+
+final class ValueGetter
+{
+    /**
+     * @param array<int|string, mixed> $raw
+     *
+     * @return mixed
+     *
+     * @template T
+     * @psalm-param array<int|string, T> $raw
+     * @psalm-return T
+     */
+    public static function first(array $raw)
+    {
+        $result = array_shift($raw);
+        assert($result !== null);
```

Comment:
Well it won't ever happen if user won't just pass random data to it ... if they use the data from snmp transport, there will always be at least a single row in that result

Reply:
Sure but what if he won't. Or somehow cripple the result on the way. Here you should apply your approach https://github.com/simPod/PHP-SNMP/pull/7#discussion_r436551617 because there's no way to handle this through types.

Reply:
ok, why not